bitcoin/src/address.rs

```diff
@@ -112,7 +114,7 @@ impl fmt::Display for Error {
             Error::ExcessiveScriptSize => write!(f, "script size exceed 520 bytes"),
             Error::UnrecognizedScript => write!(f, "script is not a p2pkh, p2sh or witness program"),
             Error::UnknownAddressType(ref s) => write!(f, "unknown address type: '{}' is either invalid or not supported in rust-bitcoin", s),
-	    Error::NetworkValidation { required, found } => write!(f, "address's network {} is different from required {}", found, required),
+	        Error::NetworkValidation { required, found, ref address } => write!(f, "address {:?} belongs to network {} which is different from required {}", address, found, required),
```

Comment:
In afd231e83ef8c62700da1ce86153b1c595195c73:

We should be able to use `{}` rather than `{:?}`. (If not, this is a missing impl in `Address<NetworkUnchecked>` since IIRC we intended that you be able to at least round-trip serialize an unchecked address.)

Reply:
I don't think we did conclude that since there's machinery to handle checked/unchecked in `Debug`. But we should use `fmt_internal` instead to not have the wrapping `Address<NetworkUnchecked>(...)` around the address.

Reply:
Ah, yeah, good point. Using `Debug` will provide some extra info. It just makes me a bit nervous to see it used in user-facing code. Maybe a comment is warranted, even when using `fmt_internal`, to explain what we're doing.

Reply:
> But we should use fmt_internal instead to not have the wrapping Address<NetworkUnchecked>(...) around the address.

Fixed this. Also added an unit test to test the display